data_science/palette/notebooks/00 - RGB, LAB, and human colour perception.ipynb

```diff
@@ -0,0 +1,133 @@
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "The red, green, and blue channels available to us in RGB space are ideally suited for representing colour on pixelated screens. However, our goal is to represent the _differences_ between colours, and RGB isn't ideal for this. It's now [pretty well established](https://en.wikipedia.org/wiki/Color_difference) that euclidian distances in RGB space are a bad representation of the distances that our eyes see.\n",
+    "By stretching the RGB dimensions by different amounts, we can better approximate that difference:\n",
+    "\n",
+    "${\\displaystyle {\\sqrt {2\\times \\Delta R^{2}+4\\times \\Delta G^{2}+3\\times \\Delta B^{2}}}}$\n",
+    "\n",
+    "Again, here's the python"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "def colour_distance(colour_1, colour_2):\n",
+    "    r_1, g_1, b_1 = colour_1\n",
+    "    r_2, g_2, b_2 = colour_2\n",
+    "\n",
+    "    return (2 * (r_1 - r_2) ** 2 +\n",
+    "            4 * (g_1 - g_2) ** 2 +\n",
+    "            3 * (b_1 - b_2) ** 2) ** 0.5"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "We can improve further by adding some extra weirdness to the red and blue channels at the end\n",
+    "\n",
+    "${\\displaystyle \\Delta C={\\sqrt {2\\times \\Delta R^{2}+4\\times \\Delta G^{2}+3\\times \\Delta B^{2}+{{{\\bar {r}}\\times (\\Delta R^{2}-\\Delta B^{2})} \\over {256}}}}}$\n",
+    "\n",
+    "Here it is in python"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "def colour_distance(colour_1, colour_2):\n",
+    "    r_1, g_1, b_1 = colour_1\n",
+    "    r_2, g_2, b_2 = colour_2\n",
+    "    \n",
+    "    d_r_sq = (r_1 - r_2) ** 2\n",
+    "    d_g_sq = (g_1 - g_2) ** 2\n",
+    "    d_b_sq = (b_1 - b_2) ** 2\n",
+    "    mean_r = (r_1 + r_2) / 2\n",
+    "    \n",
+    "    d_c_sq = (2 * d_r_sq + 4 * d_g_sq + 3 * d_b_sq +\n",
+    "              (mean_r * (d_r_sq - d_b_sq) / 256))\n",
+    "\n",
+    "    return d_c_sq ** 0.5"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "The most general and efficient approach (as far as I know) is to transform the entire image's RGB coordinates into a new space. The CIE produced [CIELAB](https://en.wikipedia.org/wiki/CIELAB_color_space#CIELAB) to better approximate human perception. The axes of the space are as follows\n",
+    "\n",
+    "> The three coordinates of CIELAB represent the lightness of the color (`L` = 0 yields black and `L` = 100 indicates diffuse white; specular white may be higher), its position between red/magenta and green (`a`, negative values indicate green while positive values indicate magenta) and its position between yellow and blue (`b`, negative values indicate blue and positive values indicate yellow).\n",
```

Comment:
I know you’re quoting Wikipedia, but breaking up these sentences into bullets would make it much easier.